Bluetooth: Fixing issue with adv_stop

martintv opened the pull request:
If adv stop was called form connected callback,
it would continue and unref the wrong conn object
if there where 2 connectable advertisers running,
but with only one 1 id configured.


This fixes the issue because with this change the bt_le_ext_adv_stop takes this exit https://github.com/zephyrproject-rtos/zephyr/blob/7ed6775d02bd54b77727a42c7695a37b66be24d6/subsys/bluetooth/host/adv.c#L1638

instead of continueing and deleiting the object of the other connectable advertiser, while failing to stop the advertiser in the controller (Because the controller accepts stoping an advertiser that is already stoped, but termineted event is not given yet)

martintv committed:
Bluetooth: Fix issue with adv_stop

If adv stop was called form connected callback,
it would continue and unref the wrong conn object
if there where 2 connectable advertisers running,
but with only one 1 id configured.

Fixes #52196

Signed-off-by: Martin Tverdal <martin.tverdal@nordicsemi.no>

jhedberg commented on subsys/bluetooth/host/adv.c:
```diff
@@ -1930,7 +1930,7 @@ void bt_hci_le_adv_set_terminated(struct net_buf *buf)
 	struct bt_le_ext_adv *adv;
 	uint16_t conn_handle;
-#if (CONFIG_BT_ID_MAX > 1) && (CONFIG_BT_EXT_ADV_MAX_ADV_SET > 1)
+#if (CONFIG_BT_MAX_CONN > 0) && (CONFIG_BT_EXT_ADV_MAX_ADV_SET > 1)
```
`BT_MAX_CONN` has a specified range with a minimum value of `1`, so testing for `> 0` will always be true if the macro is defined. Is it always guaranteed to be defined here, i.e. is `CONFIG_BT_CONN` always enabled in this part of the code? If yes, then just drop the first condition the if statement. If no, then the condition should read `#if defined(CONFIG_BT_CONN) && ...`.

jhedberg commented on subsys/bluetooth/host/adv.c:
```diff
@@ -1949,13 +1949,13 @@ void bt_hci_le_adv_set_terminated(struct net_buf *buf)
 	(void)bt_le_lim_adv_cancel_timeout(adv);
 
-#if (CONFIG_BT_ID_MAX > 1) && (CONFIG_BT_EXT_ADV_MAX_ADV_SET > 1)
+#if (CONFIG_BT_MAX_CONN > 0) && (CONFIG_BT_EXT_ADV_MAX_ADV_SET > 1)
```
Same here, and in three other places (if I counted correctly), which I won't separately point out.

jhedberg reviewed (changes requested):
See my inline comments. Also a minor thing about the commit message: please use the imperative form of the verb to be consistent with the commit history, i.e. "Fix" instead of "Fixing".

martintv commented:
> See my inline comments. Also a minor thing about the commit message: please use the imperative form of the verb to be consistent with the commit history, i.e. "Fix" instead of "Fixing".

Done

martintv commented:
> I only updated the commit message to include "Fixes #52196", and rebased on main.

Sorry, I forcepushed from wrong folder locally previously, so CI failed again. Fixed now, and now the only change relative to when you approved previously is a rebase to main, and updated commitmessage.
